routers/repo/view.go

```diff
@@ -342,6 +342,21 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		ctx.Data["IsAudioFile"] = true
 	case base.IsImageFile(buf):
 		ctx.Data["IsImageFile"] = true
+	default:
+		if fileSize >= setting.UI.MaxDisplayFileSize {
+			ctx.Data["IsFileTooLarge"] = true
+			break
+		}
+
+		d, _ := ioutil.ReadAll(dataRc)
+		buf = append(buf, d...)
+
+		if markupType := markup.Type(blob.Name()); markupType != "" {
```

Comment:
One question... what happens with buf if this condition is not met (i.e. `markupType == ""`)? Is it OK to append `d` to it anyway?

Reply:
In the case that `markupType == ""` or said another way; the file is not of a type recognised by gitea, or of an extension configured by the user in app.ini then the switch will exit without action. Which is what it did before it had the `default` case added in this PR. 

With that in mind, perhaps it would be better to have the code like this

``` suggestion
		if fileSize >= setting.UI.MaxDisplayFileSize {
			ctx.Data["IsFileTooLarge"] = true
			break
		}
		
		if markupType := markup.Type(blob.Name()); markupType != "" {
			d, _ := ioutil.ReadAll(dataRc)
			buf = append(buf, d...)
			ctx.Data["IsMarkup"] = true
			ctx.Data["MarkupType"] = markupType
			ctx.Data["FileContent"] = string(markup.Render(blob.Name(), buf, path.Dir(treeLink), ctx.Repo.Repository.ComposeMetas()))
		}
```

That way the read is only performed if needed (`markupType != "" `), rather than in all cases than where `fileSize >= setting.UI.MaxDisplayFileSize` is not `true`

Reply:
That sounds good.

Reply:
@guillep2k Changes made, thank you. Build passing.